PWGJE/EMCALJetTasks/AliAnalysisTaskRhoSparseSV.cxx

```diff
@@ -0,0 +1,207 @@
+#include "AliLog.h"
+#include "AliRhoParameter.h"
+#include "AliJetContainer.h"
+
+ClassImp(AliAnalysisTaskRhoSparseSV)
+
+//________________________________________________________________________
+AliAnalysisTaskRhoSparseSV::AliAnalysisTaskRhoSparseSV() : 
+  AliAnalysisTaskRhoBase("AliAnalysisTaskRhoSparseSV"),
+  fNExclLeadJets(0),
+  fRhoCMS(0),
+  fHistOccCorrvsCent(0)
+{
+  // Constructor.
+}
+
+//________________________________________________________________________
+AliAnalysisTaskRhoSparseSV::AliAnalysisTaskRhoSparseSV(const char *name, Bool_t histo) :
+  AliAnalysisTaskRhoBase(name, histo),
+  fNExclLeadJets(0),
+  fRhoCMS(0),
+  fHistOccCorrvsCent(0)
+{
+  // Constructor.
+}
+
+//________________________________________________________________________
+void AliAnalysisTaskRhoSparseSV::UserCreateOutputObjects()
+{
+  if (!fCreateHisto) return;
+
+  AliAnalysisTaskRhoBase::UserCreateOutputObjects();
+  
+  fHistOccCorrvsCent = new TH2F("OccCorrvsCent", "OccCorrvsCent", 101, -1, 100, 2000, 0 , 2);
+  fOutput->Add(fHistOccCorrvsCent);
+}
+
+//________________________________________________________________________
+Bool_t AliAnalysisTaskRhoSparseSV::IsJetOverlapping(AliEmcalJet* jet1, AliEmcalJet* jet2)
+{
+  for (Int_t i = 0; i < jet1->GetNumberOfTracks(); ++i)
+  {
+    Int_t jet1Track = jet1->TrackAt(i);
+    for (Int_t j = 0; j < jet2->GetNumberOfTracks(); ++j)
+    {
+      Int_t jet2Track = jet2->TrackAt(j);
+      if (jet1Track == jet2Track)
+        return kTRUE;
+    }
+  }
+  return kFALSE;
+}
+
+//________________________________________________________________________
+Bool_t AliAnalysisTaskRhoSparseSV::IsJetSignal(AliEmcalJet* jet)
+{
+  if(jet->Pt()>5){
+      return kTRUE;
+  }else{
+    return kFALSE;
+  }
+}
+
+
+//________________________________________________________________________
+Bool_t AliAnalysisTaskRhoSparseSV::Run() 
+{
+  // Run the analysis.
+
+  fOutRho->SetVal(0);
+  if (fOutRhoScaled)
+    fOutRhoScaled->SetVal(0);
+
+  if (!fJets)
+    return kFALSE;
+
+  const Int_t Njets = fJets->GetEntries();
+
+  AliJetContainer *sigjets = static_cast<AliJetContainer*>(fJetCollArray.At(1));
+
+  Int_t NjetsSig = 0;
+  if (sigjets) NjetsSig = sigjets->GetNJets();
+
+  Int_t maxJetIds[]   = {-1, -1};
+  Float_t maxJetPts[] = { 0,  0};
+
+  if (fNExclLeadJets > 0) {
+    for (Int_t ij = 0; ij < Njets; ++ij) {
+      AliEmcalJet *jet = static_cast<AliEmcalJet*>(fJets->At(ij));
+      if (!jet) {
+	AliError(Form("%s: Could not receive jet %d", GetName(), ij));
+	continue;
+      } 
+
+      if (!AcceptJet(jet))
+        continue;
+
+      if (jet->Pt() > maxJetPts[0]) {
+	maxJetPts[1] = maxJetPts[0];
+	maxJetIds[1] = maxJetIds[0];
+	maxJetPts[0] = jet->Pt();
+	maxJetIds[0] = ij;
+      } else if (jet->Pt() > maxJetPts[1]) {
+	maxJetPts[1] = jet->Pt();
+	maxJetIds[1] = ij;
+      }
+    }
+    if (fNExclLeadJets < 2) {
+      maxJetIds[1] = -1;
+      maxJetPts[1] = 0;
+    }
+  }
+
+  static Double_t rhovec[999];

```

Comment:
Please avoid such constructs (static large block of memory) as it is excessive use of resources and creates troubles in parallelized environments. Better use std::vector, they are quite efficient.